go/expr/go.mod

```diff
@@ -0,0 +1,8 @@
+module cel.dev/expr
```

Comment:
@TristonianJones Thoughts on the directory structure? In case if you wanted to serve any other generated golang protos from here

Reply:
I don't have a lot of opinions on the directory structure; however, I also want to cut back on Go's reliance on proto in general, so I think having this single directory is pretty perfect for making sure that continues to be the case.